packages/ember-metal/lib/chains.js

```diff
@@ -40,7 +40,7 @@ ChainWatchers.prototype = {
   remove(key, node) {
     let nodes = this.chains[key];
     if (nodes) {
-      for (var i = 0, l = nodes.length; i < l; i++) {
+      for (var i = 0; i < nodes.length; i++) {
         if (nodes[i] === node) {
           nodes.splice(i, 1);
```

Comment:
On a second look, this shouldn't need to cache length as we break the loop on the next line, might be confusing though.